### PR TITLE
NIFI-3580 - configure TLS cipher suites

### PR DESCRIPTION
**Pull request by greyp9**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by greyp9** [comment]
Bad upstream rebase in Previous PR.
https://github.com/apache/nifi/pull/5008

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 1]
```diff
@@ -3494,6 +3516,14 @@ Providing three total network interfaces, including  `nifi.web.http.network.inte
 |`nifi.web.https.host`|The HTTPS host. It is blank by default.
 |`nifi.web.https.port`|The HTTPS port. It is blank by default. When configuring NiFi to run securely, this port should be configured.
 |`nifi.web.https.port.forwarding`|Same as `nifi.web.http.port.forwarding`, but with HTTPS for secure communication. It is blank by default.
+|`nifi.web.https.includeciphersuites`|Cipher suites used to initialize the SSLContext of the Jetty https port.  If unspecified, the runtime SSLContext defaults are used.
```
Comment (exceptionfactory) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc:3519]: Looks like this property name needs to be updated, also recommend changing `https` to uppercase `HTTPS`:
```suggestion
|`nifi.web.https.ciphersuites.include`|Cipher suites used to initialize the SSLContext of the Jetty HTTPS port.  If unspecified, the runtime SSLContext defaults are used.
```

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 2]
```diff
@@ -3494,6 +3516,14 @@ Providing three total network interfaces, including  `nifi.web.http.network.inte
...
+|`nifi.web.https.excludeciphersuites`|Cipher suites that may not be used by an SSL client to establish a connection to Jetty.  If unspecified, the runtime SSLContext defaults are used.
```
Comment (exceptionfactory) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc:3520]: This property name also needs to be updated:
```suggestion
|`nifi.web.https.ciphersuites.exclude`|Cipher suites that may not be used by an SSL client to establish a connection to Jetty.  If unspecified, the runtime SSLContext defaults are used.
```

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java [hunk 3]
```diff
@@ -1012,6 +1016,28 @@ protected static void configureSslContextFactory(SslContextFactory.Server contex
         contextFactory.setIncludeProtocols(TlsConfiguration.getCurrentSupportedTlsProtocolVersions());
         contextFactory.setExcludeProtocols("TLS", "TLSv1", "TLSv1.1", "SSL", "SSLv2", "SSLv2Hello", "SSLv3");
 
+        // on configuration, replace default application cipher suites with those configured
+        final String includeCipherSuitesProps = props.getProperty(NiFiProperties.WEB_HTTPS_CIPHERSUITES_INCLUDE);
+        if (StringUtils.isNotEmpty(includeCipherSuitesProps)) {
+            final String[] includeCipherSuitesRuntime = contextFactory.getIncludeCipherSuites();
+            final String[] includeCipherSuites = includeCipherSuitesProps.split(REGEX_SPLIT_PROPERTY);
+            logger.info("Replacing include cipher suites with configuration; runtime = {}, raw property = {}, parsed property = {}.",
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java:1024]: Do you think it is likely that the raw property will be that different from the parsed property?  It seems like the only difference should be the number of spaces, in which case, logging both doesn't seem very useful.

**Review by exceptionfactory — COMMENTED** [review]
Thanks for making the updates @greyp9, noted a few additional very minor details, otherwise looks good.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java (hunk 3 above)
Comment (greyp9) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java:1024]: I'm worried about edge cases, but may be overly cautious.  Will change to logging just the parsed value.

**Review by exceptionfactory — APPROVED** [review]
Thanks for your work on this @greyp9.  Final round of updates look good. +1 Merging.

**Pull request closed without merging**